domains/runtime/auto-id/src/lib.rs

```diff
@@ -609,20 +625,67 @@ fn check_transaction_and_do_pre_dispatch_inner(
     // as that will add the side effect of SignedExtension in the storage buffer
     // which would help to maintain context across multiple transaction validity check against same
     // runtime instance.
-    match xt.signed {
+    match xt.format {
         // signed transaction
-        Some((account_id, extra)) => extra
-            .pre_dispatch(&account_id, &xt.function, &dispatch_info, encoded_len)
-            .map(|_| ()),
+        ExtrinsicFormat::Signed(account_id, extra) => {
+            let origin = RuntimeOrigin::signed(account_id);
+            let implication = TxBaseImplication(xt.function.clone());
+            let (_, nonce_val, origin) = frame_system::CheckNonce::<Runtime>::validate(
+                &extra.5,
+                origin,
+                &xt.function,
+                &dispatch_info,
+                encoded_len,
+                (),
+                &implication,
+                TransactionSource::External,
+            )?;
+            let (_, weight_val) =
+                frame_system::CheckWeight::<Runtime>::do_validate(&dispatch_info, encoded_len)?;
+
+            let (_, transaction_payment_val, origin) =
+                pallet_transaction_payment::ChargeTransactionPayment::<Runtime>::validate(
+                    &extra.7,
+                    origin,
+                    &xt.function,
+                    &dispatch_info,
+                    encoded_len,
+                    (),
+                    &implication,
+                    TransactionSource::External,
+                )?;
```

Comment:
Not sure why only manually validate 3 extensions here, there are more extensions in the `SignedExtra`, can we do `<SignedExtra as TransactionExtension<RuntimeCall>>::validate(...)`?

Reply:
I think you are looking at a older commit which I resolved later in later commits